lib/web_ui/test/engine/platform_views/content_manager_test.dart

```diff
@@ -0,0 +1,170 @@
+    });
+
+    group('knowsViewType', () {
+      test('recognizes viewTypes after registering them', () async {
+        expect(contentManager.knowsViewType(viewType), isFalse);
+
+        contentManager.registerFactory(viewType, (int id) => html.DivElement());
+
+        expect(contentManager.knowsViewType(viewType), isTrue);
+      });
+    });
+
+    group('knowsViewId', () {
+      test('recognizes viewIds after *rendering* them', () async {
+        expect(contentManager.knowsViewId(viewId), isFalse);
+
+        contentManager.registerFactory(viewType, (int id) => html.DivElement());
+        contentManager.renderContent(viewType, viewId, null);
+
+        expect(contentManager.knowsViewId(viewId), isTrue);
+      });
+
+      test('forgets viewIds after clearing them', () {
+        contentManager.registerFactory(viewType, (int id) => html.DivElement());
+        contentManager.renderContent(viewType, viewId, null);
+
+        expect(contentManager.knowsViewId(viewId), isTrue);
+
+        contentManager.clearPlatformView(viewId);
+
+        expect(contentManager.knowsViewId(viewId), isFalse);
+      });
+    });
+
+    group('registerFactory', () {
+      test('does NOT re-register factories', () async {
+        contentManager.registerFactory(
+            viewType, (int id) => html.DivElement()..id = 'pass');
+        // this should be rejected
+        contentManager.registerFactory(
+            viewType, (int id) => html.SpanElement()..id = 'fail');
+
+        final html.Element contents =
+            contentManager.renderContent(viewType, viewId, null);
+
+        expect(contents.querySelector('#pass'), isNotNull);
+        expect(contents.querySelector('#fail'), isNull,
+            reason: 'Factories cannot be overridden once registered');
+      });
+    });
+
+    group('renderSlot', () {
+      test(
+          'can render slot, even for views that might have never been rendered before',
+          () async {
+        final html.Element slot = contentManager.renderSlot(viewId);
+        expect(slot, isNotNull);
+        expect(slot.querySelector('slot'), isNotNull);
+      });
+
+      test('rendered markup contains required attributes', () async {
+        final html.Element slot = contentManager.renderSlot(viewId);
+        expect(slot.style.pointerEvents, 'auto',
+            reason:
+                'Should re-enable pointer events for the contents of the view.');
+        final html.Element innerSlot = slot.querySelector('slot')!;
+        expect(innerSlot.getAttribute('name'), contains('$viewId'),
+            reason:
+                'The name attribute of the inner SLOT tag must refer to the viewId.');
+      });
+
+      test('returns cached instances of already-rendered slots', () async {
+        final html.Element firstSlot = contentManager.renderSlot(viewId);
+        final html.Element sameSlot = contentManager.renderSlot(viewId);
+
+        expect(firstSlot, sameSlot);
```

Comment:
nit: you can use `same` from `package:test` here which I think explicitly uses `identical`

Reply:
Ah yes, by not passing anything I'm settling to `equals`. `identical` is what I wanted. I did the same thing for the `renderContent` test too.